src/main/java/org/gaul/s3proxy/S3Proxy.java

```diff
@@ -368,15 +378,15 @@ public boolean equals(Object object) {
                             that.v4MaxNonChunkedRequestSize) &&
                     Objects.equals(this.ignoreUnknownHeaders,
                             that.ignoreUnknownHeaders) &&
-                    Objects.equals(this.corsAllowAll, that.corsAllowAll);
+                    this.corsRules.equals(that.corsRules);
         }
 
         @Override
         public int hashCode() {
             return Objects.hash(endpoint, secureEndpoint, keyStorePath,
                     keyStorePassword, virtualHost, servicePath,
                     v4MaxNonChunkedRequestSize, ignoreUnknownHeaders,
-                    corsAllowAll);
+                    corsRules.hashCode());
```

Comment:
Should you pass `corsRules` instead of its `hashCode`?

Reply:
done